docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+9. Procedure that performs resolving DID will fallback to old wallet structure to allow using existing crypto materials with the new approach
+10. The old (current) DID API will be deprecated
+11. The new DID API will be created to allow:
+     - Registering of DID methods
+     - Creation of DIDs
+     - Building of DID Docs and DID Doc deltas with builder pattern
+     - Publishing of DID Doc and DID Doc deltas
+     - Resolving of DID Docs
+
+## Libindy API
+
+### DID resolver configuration
+
+#### Create DID Resolver instance
+
+```Rust
+// Create DID resolver instance
+//
+// By default supports peer method only. To support additional did methods
+// indy_register_indy_did_method or indy_register_did_method can be called.
+//
+// options: - some options like cache policies
+// returns - did_resolver_handle: handler ot DID Resolver instance
+pub fn indy_create_did_resolver(options) -> Future<did_resolver_handle>
+```
+
+#### Register did method based on Indy pool with did resolver
+
+```Rust
+// Register did method based on Indy pool with did resolver.
+//
+// did_resolver_handle: resolver handle
+// method: method name, for example, 'sov'
+// pool_handle: pool_handle
+// options: some options
+// returns - ()
+pub fn indy_register_indy_did_method(did_resolver_handle,
+                                     method,
+                                     pool_handle,
+                                     options) -> Future<()>
+```
+
+#### Register custom did method with did resolver
+
+```Rust
+// Register custom did method with did resolver.
+//
+// did_resolver_handle: resolver handle
+// method: method name, for example, 'custom'
+// ..callbacks: list of callbacks that define method behaviour
+// options: some options
+// returns - ()
+pub fn indy_register_did_method(did_resolver_handle,
+                                method,
+                                ..callbacks,
+                                options) -> Future<()>
+```
+
+#### Close did resolver and releases all related resources
+
+```Rust
+// Close did resolver and releases all related resources.
+//
+// did_resolver_handle: resolver handle
+// returns - ()
+pub fn indy_close_did_resolver(did_resolver_handle) -> Future<()>
+```
+
+### Resolving DID Doc
+
+#### Resolve DID Doc for DID
+
+```Rust
+// Resolve DID Doc for DID.
+//
+// did_resolver_handle: resolver handle
+// did: did to resolve
+// options: some options, like cache behaviour
+// returns - did_doc: resolved DID Doc
+pub fn indy_resolve_did_doc(did_resolver_handle,
+                            did,
+                            options) -> Future<did_doc>
+```
+
+### DID and DID keys creation
+
+#### Creates a new DID
+
+```Rust
+// Create a new DID.
+//
+// Note: It doesn't create any records in the wallet.
+//
+// did_resolver_handle: resolver handle
+// method: did method to use
+// options: some options
+// returns - did: Generated DID
+pub fn indy_create_did(did_resolver_handle,
```

Comment:
When is this method needed alone? Can it have functionlity of `indy_create_did_key`

Reply:
I think it makes sense to have this method as a separate one because of the following:
1) DID can be generated independent of the key. It can be just a UUID, not necessary a part of a public key
2) It makes the API more clear and clean from my opinion, especially if we want to create multiple keys for a DID in one builder.
3) I think there can be use cases when we want just DID, and keys will be generated latter.